packages/plugin-core/src/commands/RefactorHierarchyV2.ts

```diff
@@ -114,45 +229,61 @@ export class RefactorHierarchyCommandV2 extends BasicCommand<
     panel.webview.html = md.render(content);
   }
 
-  async execute(opts: CommandOpts): Promise<any> {
-    const ctx = "RefactorHierarchy:execute";
-    const { match, replace } = _.defaults(opts);
-    this.L.info({ ctx, opts, msg: "enter" });
-    const ext = getExtension();
-    const { engine } = getDWorkspace();
-    const notes = engine.notes;
-    const re = new RegExp(`(.*)(${match})(.*)`);
-    const candidates = _.filter(notes, (n) => {
-      if (n.stub) {
-        return false;
+  getCapturedEntries(opts: {
+    scope: NoteLookupProviderSuccessResp | undefined;
+    matchRE: RegExp;
+    engine: DEngineClient;
+  }) {
+    const { scope, matchRE, engine } = opts;
+    const { notes, schemas, vaults, wsRoot } = engine;
+
+    const capturedEntries: NoteQuickInput[] = [];
+
+    const scopedItems = _.isUndefined(scope)
+      ? _.toArray(notes).map((note: DNodeProps) => {
```

Comment:
this can be very expensive to do (eg. our workspace is +10k notes). instead of converting everything to `QuickInput`, can we do the reverse? Convert `QuickInput` items to notes?

Reply:
fixed